core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java

```diff
@@ -60,12 +66,51 @@ public HadoopTables(Configuration conf) {
   public Table load(String location) {
     TableOperations ops = newTableOps(location);
     if (ops.current() == null) {
-      throw new NoSuchTableException("Table does not exist at location: " + location);
+      // try to resolve a metadata table, which we encode as URI fragments
+      // e.g. hdfs:///warehouse/my_table#snapshots
+      int hashIndex = location.lastIndexOf('#');
+      if (hashIndex != -1 && location.length() - 1 != hashIndex) {
```

Comment:
Can we use regex pattern instead?

Reply:
@chenjunjiedada can you elaborate on why a regex would be better?

Reply:
@xabriel, IMHO, the regex can prevent a much larger range of invalid input for the public API which usage is not under control.

Reply:
Thanks @chenjunjiedada. I'd rather keep simple code if there is no concrete example of an issue or security concern?

Consider existing code in `IcebergSource` that does a simple check on the string to see if it looks like a path: https://github.com/apache/incubator-iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java#L109-L111. You could do it with a regex instead, but so much simpler to just check for `contains("/")`.

@rdblue and @aokolnychyi, any preference?

Reply:
I don't have a strong preference here. We use `lastIndexOf` in a few places. For example, `TableMetadataParser` and `BaseMetastoreTableOperations`, so I am OK with the way it is.